src/components/components/Component.js

```diff
@@ -89,24 +126,29 @@ export default class Component extends React.Component {
     }
   };
 
+  selectOption = objName => {
+    const value = this.state.nameToLinkMap[objName];
+    updateEntity.apply(this, [this.props.entity, this.props.name, value]);
+  }
+
   /**
    * Render propert(ies) of the component.
    */
   renderPropertyRows = () => {
     const componentData = this.props.component;
 
     if (isSingleProperty(componentData.schema)) {
-      const componentName = this.props.name;
-      const schema = AFRAME.components[componentName.split('__')[0]].schema;
       return (
-        <PropertyRow
-          key={componentName}
-          name={componentName}
-          schema={schema}
-          data={componentData.data}
-          componentname={componentName}
-          isSingle={true}
-          entity={this.props.entity}
+        <Select
+          id="gltfModelSelect"
+          ref="select"
+          options={this.state.nameList}
```

Comment:
The options aren't rendered properly (can't see the text). @ahmedhamodi and @amoliksingh can you work together and fix the CSS to make it work properly?